FindTeamer/ViewsContent/Authentication/SignUpView.swift

```diff
@@ -6,107 +6,93 @@
         Button(action: registerClicked) {
             ButtonContent(text: "Let's Go!")
         }.opacity(buttonOpacity)
-        .alert(isPresented: $presentAlert) {
-            switch activeAlert {
-            case .first:
-                return Alert (
-                    title: Text(alertModel.title),
-                    message: Text(alertModel.message),
-                    dismissButton: .default(Text(alertModel.buttonNaming)) {
-                        presentationMode.wrappedValue.dismiss()
-                    })
-            case .second:
-                return Alert (
-                    title: Text(alertModel.title),
-                    message: Text(alertModel.message),
-                    dismissButton: .default(Text(alertModel.buttonNaming)))
+            .alert(isPresented: $presentAlert) {
+                switch activeAlert {
+                case .first:
+                    return Alert(
+                        title: Text(alertModel.title),
+                        message: Text(alertModel.message),
+                        dismissButton: .default(Text(alertModel.buttonNaming)) {
+                            presentationMode.wrappedValue.dismiss()
+                        })
+                case .second:
+                    return Alert(
+                        title: Text(alertModel.title),
+                        message: Text(alertModel.message),
+                        dismissButton: .default(Text(alertModel.buttonNaming)))
+                }
             }
-        }
     }
-    
     var buttonOpacity: Double {
-        return formModel.isFormValid &&
-                formModel.errorMessages.count == 0 ? 1 : 0.5
+        return signUpViewModel.isFormValid &&
+        signUpViewModel.validator.errorMessages.isEmpty ? 1 : 0.5
     }
-    
     func registerClicked() {
-        self.errorArray = ""
-        if self.formModel.isFormValid && self.formModel.errorMessages.count == 0 {
-            signUpManager.createUser(email: self.formModel.userEmail, password: self.formModel.userPassword, completionBlock: { (success) in
-                    presentAlert = true
-                    if (success) {
-                        activeAlert = .first
-                        alertModel.title = "Success"
-                        alertModel.message = "Registration complete!"
-                    } else {
-                        activeAlert = .second
-                        alertModel.title = "Error"
-                        alertModel.message = "Registration failed. Please try again!"
-                        alertModel.buttonNaming = "OK"
+        self.errors = ""
+        if self.signUpViewModel.isFormValid && self.signUpViewModel.validator.errorMessages.isEmpty {
+            signUpViewModel.createUser(email: self.signUpViewModel.modelState.userEmail, password: self.signUpViewModel.modelState.userPassword, completionBlock: { (success) in
+                presentAlert = true
+                if (success) {
```

Comment:
This check should be in ViewModel and ViewModel should tell View whether we need to show Alert or not and content of the alert